python/lsst/pipe/analysis/utils.py

```diff
@@ -140,15 +140,19 @@ def __call__(self, catalog):
 
 class MagDiffMatches(object):
     """Functor to calculate magnitude difference for match catalog"""
-    def __init__(self, column, colorterm, zp=27.0, unitScale=1.0):
+    def __init__(self, column, colorterm, filterName, zp=27.0, unitScale=1.0):
         self.column = column
         self.colorterm = colorterm
+        self.filterName = filterName
         self.zp = zp
         self.unitScale = unitScale
     def __call__(self, catalog):
-        ref1 = -2.5*np.log10(catalog["ref_" + self.colorterm.primary + "_flux"])
-        ref2 = -2.5*np.log10(catalog["ref_" + self.colorterm.secondary + "_flux"])
-        ref = self.colorterm.transformMags(ref1, ref2)
+        if self.colorterm is not None:
+            ref1 = -2.5*np.log10(catalog["ref_" + self.colorterm.primary + "_flux"])
+            ref2 = -2.5*np.log10(catalog["ref_" + self.colorterm.secondary + "_flux"])
+            ref = self.colorterm.transformMags(ref1, ref2)
+        else:
+            ref = self.zp - 2.5*np.log10(catalog["ref_" + self.filterName + "_flux"])
```

Comment:
Instead of making any changes to this functor, I think it would be much simpler to pass in a null colorterm.